aquadoggo/src/http/service.rs

```diff
@@ -38,7 +40,12 @@ pub fn build_server(http_context: HttpServiceContext) -> Router {
 }
 
 /// Start HTTP server.
-pub async fn http_service(context: Context, signal: Shutdown, tx: ServiceSender) -> Result<()> {
+pub async fn http_service(
+    context: Context,
+    signal: Shutdown,
+    tx: ServiceSender,
+    tx_ready: oneshot::Sender<()>,
```

Comment:
Maybe give the sender also an own type like you already do for the receiver (`ServiceReady`)? I think that would also ready easier as well.

Reply:
Like this?